sonarts-core/src/rules/preferPromiseShorthandRule.ts

```diff
@@ -0,0 +1,121 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+import * as tslint from "tslint";
+import * as ts from "typescript";
+import { SonarRuleMetaData } from "../sonarRule";
+import { SonarRuleVisitor } from "../utils/sonarAnalysis";
+import {
+  isIdentifier,
+  isFunctionLikeDeclaration,
+  isBlock,
+  isExpressionStatement,
+  isCallExpression,
+} from "../utils/nodes";
+
+export class Rule extends tslint.Rules.AbstractRule {
+  public static metadata: SonarRuleMetaData = {
+    ruleName: "prefer-promise-shorthand",
+    description: "Shorthand promises should be used",
+    rationale: tslint.Utils.dedent``,
+    optionsDescription: "",
+    options: null,
+    rspecKey: "RSPEC-4634",
+    type: "maintainability",
+    typescriptOnly: true,
+  };
+
+  public apply(sourceFile: ts.SourceFile): tslint.RuleFailure[] {
+    return new Visitor(this.getOptions().ruleName).visit(sourceFile).getIssues();
+  }
+}
+
+function message(value: ts.Node, action: PromiseAction) {
+  return `Replace this trivial promise with "Promise.${action}(${value.getText()})".`;
+}
+
+class Visitor extends SonarRuleVisitor {
+  protected visitNewExpression(node: ts.NewExpression) {
+    const executor = getPromiseExecutor(node);
+    if (executor) {
+      if (isFunctionLikeDeclaration(executor) && executor.body) {
+        const { parameters, body } = executor;
+        const [resolveParameterDeclaration, rejectParameterDeclaration] = parameters;
+        const resolveParameterName = getParameterName(resolveParameterDeclaration);
+        const rejectParameterName = getParameterName(rejectParameterDeclaration);
+
+        let bodyExpression = getOnlyBodyExpression(body);
+
+        if (bodyExpression && isCallExpression(bodyExpression)) {
+          const { expression: callee, arguments: callArguments } = bodyExpression;
+          if (isIdentifier(callee)) {
+            const action = getPromiseAction(callee.text, resolveParameterName, rejectParameterName);
+            if (action && callArguments.length === 1) this.addIssue(node, message(callArguments[0], action));
+          }
+        }
+      }
+    }
+    super.visitNewExpression(node);
+  }
+}
+
+function getPromiseExecutor(node: ts.NewExpression) {
+  if (
+    isIdentifier(node.expression) &&
+    node.expression.text === "Promise" &&
+    node.arguments &&
+    node.arguments.length === 1
+  ) {
+    return node.arguments[0];
+  }
+}
+
+function getOnlyBodyExpression(body: ts.ConciseBody) {
+  let bodyExpression: ts.Expression | undefined;
+  if (isBlock(body)) {
+    if (body.statements.length === 1) {
+      let statement = body.statements[0];
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Identifier 'statement' is never reassigned; use 'const' instead of 'let'. ([typescript:S3353](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=typescript%3AS3353))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=sonarts&pullRequest=639&issues=AWNkE6wJ1_DDLGcJA7A2&open=AWNkE6wJ1_DDLGcJA7A2)

[ISSUE_KEY]: AWNkE6wJ1_DDLGcJA7A2